pkg/msgconv/from-matrix.go

```diff
@@ -265,7 +264,103 @@ func (mc *MessageConverter) reuploadFileToMeta(ctx context.Context, client *mess
 			Str("mime_type", mime).
 			Bool("is_voice_clip", isVoice).
 			Msg("Failed upload metadata")
-		return nil, fmt.Errorf("%w: %w", bridgev2.ErrMediaReuploadFailed, err)
+		return 0, fmt.Errorf("%w: %w", bridgev2.ErrMediaReuploadFailed, err)
+	}
+	attachmentID := resp.Payload.RealMetadata.GetFbId()
+	if attachmentID == 0 {
+		zerolog.Ctx(ctx).Warn().RawJSON("response", resp.Raw).Msg("No fbid received for upload")
+	}
+	if attachmentID == 0 && content.MsgType == event.MsgVideo && client.Platform == types.Instagram {
+		attachmentID, err = mc.reuploadVideoToMetaFallback(ctx, client, data, mime)
+		if err != nil {
+			zerolog.Ctx(ctx).Warn().Err(err).Msg("Failed to upload attachment via Instagram Android fallback")
+			return 0, err
+		} else {
+			zerolog.Ctx(ctx).Info().Msg("Uploaded attachment via Instagram Android fallback")
+		}
+	}
+	if attachmentID == 0 {
+		return 0, fmt.Errorf("failed to upload attachment: fbid not received")
```

Comment:
This should actually also use the wrapped error

```suggestion
		return 0, fmt.Errorf("%w: fbid not received", bridgev2.ErrMediaReuploadFailed)
```